text/0000-fabric-x-next-generation.md

```diff
@@ -0,0 +1,365 @@
+* **Threshold Signatures & Traditional Policies with Pre-stored Identities:** Fabric-X will support threshold signatures for compact, efficient validation. Additionally, traditional endorsement policies will be supported, with an optimization to pre-store identities (e.g., X.509 certificate details) in a configuration block (e.g., via the "known identities" feature in the MSP folder), allowing transactions to reference these identities rather than including full certificates with each signature, thus reducing transaction size.
+* **Dependency Analysis Enables Parallel Verification and Validation:** The Coordinator constructs a dependency graph for transactions. This allows transactions without conflicts to be verified and validated in parallel by multiple signature-verifier and VC instances, significantly speeding up the commit phase. This early conflict detection and pruning (before full MVCC against the database) is key to maintaining high goodput even with conflicting transactions.
+* **Flattened Transaction Structure:** To reduce processing overhead associated with serialization/deserialization of deeply nested structures, Fabric-X adopts a flatter transaction model (see "Reference-level explanation" for details).
+* **Sequential Versioning:** Fabric-X adopts sequential versioning of key/values. This allows endorsers to predict the next version without querying the database, potentially speeding up endorsement.
+
+**Impact on Fabric Users:**
+
+* **Deployment:** Deployment scripts and tools will need to be adapted for a microservice environment (e.g., using Kubernetes or similar orchestration platforms). Each organization will deploy its set of microservices and its own distributed database instance. The minimum number of logical components per organization is ten (1 endorser, router, batcher, consenter, assembler, sidecar, coordinator, signature/policy verifier, validator-committer, and a single database node cluster), though these can be co-hosted for smaller setups. For BFT consensus in Arma, a minimum of 4 Consenter nodes (to tolerate F=1 faulty node) distributed across participating organizations is required.
+* **Configuration:** Configuration will be more granular, specific to each microservice. The genesis configuration will use the same config block format and `configtxgen` (or equivalent), with additional fields for the new ordering services. MSP, ACLs, and Policies remain conceptually the same.
+* **Debugging:** Troubleshooting will involve analyzing logs and interactions across multiple services.
+* **Performance:** Expect significantly higher throughput (demonstrated >150,000 TPS) and lower latency.
+* **Feature Set:** Be aware that Fabric-X will not initially support all HLF v2.x/v3.x features (e.g., Private Data Collections, Channels). The focus is on core transactional performance. A roadmap for reintroducing essential Fabric features will be developed post-initial release.
```

Comment:
Since you are positioning this as Fabric's evolution, I guess some initial roadmap to introduce commitment has to be introduced within the RFC and should not be postponed into the later post-initial release, undefined phase.